File: python/src/aiconfig/eval/promptfoo/run_aiconfig.py
import asyncio
import os

import openai
from dotenv import load_dotenv

from aiconfig import AIConfigRuntime
import sys
import json
from typing import Any


async def main():
    settings_path = sys.argv[1]
    settings = _load_settings(settings_path)

    question = sys.argv[2]
    prompt_name = settings["prompt_name"]
    aiconfig_path = settings["aiconfig_path"]

    load_dotenv()
    openai.api_key = os.getenv("OPENAI_API_KEY")
    runtime = AIConfigRuntime.load(aiconfig_path)

    params = {
        "the_query": question,
    }

    result = await runtime.run(prompt_name, params)
    final_output = runtime.get_output_text(prompt_name, result[0])
    print(final_output)


def _load_settings(settings_path: str) -> dict[str, Any]:
    with open(settings_path) as f:
        settings = json.load(f)
        return settings


if __name__ == "__main__":
    asyncio.run(main())

File: typescript/scripts/eval/promptfoo/run_aiconfig.ts
import { AIConfigRuntime } from "../../../lib/config";
import fs from "fs";
import path from "path";

async function main() {
  let settings_path = process.argv[2];
  let question = process.argv[3];
  let settings = _load_settings(settings_path);
  let prompt_name = settings["prompt_name"];
  let aiconfig_path = settings["aiconfig_path"];

  let fullAIConfigPath = path.join(__dirname, aiconfig_path);
  let runtime = AIConfigRuntime.load(fullAIConfigPath);
  let params = {
    the_query: question,
  };
  let result = await runtime.run(prompt_name, params);
  let final_output: string;

  let r0 = Array.isArray(result) ? result[0] : result;
  final_output = runtime.getOutputText(prompt_name, r0);
  console.log(final_output);
}

function _load_settings(settings_path: string) {
  let rawdata = fs.readFileSync(settings_path, "utf-8");
  let settings = JSON.parse(rawdata);
  return settings;
}

main();
